EnhancedMediaPlayer/MediaPlayer.swift

```diff
@@ -0,0 +1,102 @@
+import SwiftUI
+import AVKit
+
+struct MediaPlayerControllerRepresentable: UIViewControllerRepresentable {
+    private let mediaURL: URL
+    private let player = AVPlayer()
+
+    init(mediaURL: URL) {
+        self.mediaURL = mediaURL
+    }
+
+    func makeUIViewController(context: Context) -> AVPlayerViewController {
+        let playerController = AVPlayerViewController()
+        setupPlayerController(playerController)
+
+        return playerController
+    }
+
+    func updateUIViewController(_ playerController: AVPlayerViewController, context: Context) {
+        if let currentURL = playerController.player?.currentItem?.asset as? AVURLAsset,
+           currentURL.url != mediaURL {
+            setupPlayerController(playerController)
+        }
+    }
+
+    func play() {
+        player.play()
+    }
+
+    func pause() {
+        player.pause()
+    }
+
+    private func setupPlayerController(_ playerController: AVPlayerViewController) {
+        playerController.showsPlaybackControls = false
+        playerController.player = player
+        playerController.player?.replaceCurrentItem(with: AVPlayerItem(url: mediaURL))
+        playerController.player?.play()
```

Comment:
What about keeping the last state, if playing keep playing if paused keep paused, this is one of the requirements of the other functions we will implement